tests/hypothesis_strategies.py

```diff
@@ -0,0 +1,158 @@
+"""Hypothesis strategies."""
+import random
+
+import hypothesis.strategies as st
+import numpy as np
+from hypothesis import assume
+from hypothesis.extra.pandas import columns, data_frames
+
+from baybe.exceptions import NumericalUnderflowError
+from baybe.parameters.categorical import (
+    CategoricalEncoding,
+    CategoricalParameter,
+    TaskParameter,
+)
+from baybe.parameters.custom import CustomDiscreteParameter
+from baybe.parameters.numerical import (
+    NumericalContinuousParameter,
+    NumericalDiscreteParameter,
+)
+from baybe.parameters.substance import SubstanceEncoding, SubstanceParameter
+from baybe.utils import DTypeFloatNumpy
+
+_largest_lower_interval = np.nextafter(
+    np.nextafter(np.inf, 0, dtype=DTypeFloatNumpy), 0, dtype=DTypeFloatNumpy
+)
+"""
+The largest possible value for the lower end of a continuous interval such that there
+still exists a larger but finite number for the upper interval end.
+"""
+
+_decorrelate = st.one_of(
+    st.booleans(),
+    st.floats(min_value=0.0, max_value=1.0, exclude_min=True, exclude_max=True),
+)
+"""A strategy that generates decorrelation settings."""
+
+parameter_name = st.text(min_size=1)
+"""A strategy that generates parameter names."""
+
+categories = st.lists(st.text(min_size=1), min_size=2, unique=True)
+"""A strategy that generates parameter categories."""
+
+
+@st.composite
+def smiles(draw: st.DrawFn):
+    """Generate short SMILES strings."""
+    n_atoms = draw(st.integers(min_value=0, max_value=19))
+    string = "C"
+    for _ in range(n_atoms):
+        next_atom = random.choice("CNO") if string[-1] == "C" else random.choice("C")
+        string += next_atom
+    return string
+
+
+@st.composite
+def substance_data(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.substance.SubstanceParameter`."""
+    names = draw(st.lists(st.text(min_size=1), min_size=2, max_size=10, unique=True))
+    substances = draw(st.lists(smiles(), min_size=len(names), max_size=len(names)))
+    return dict(zip(names, substances))
+
+
+@st.composite
+def custom_encodings(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.custom.CustomDiscreteParameter`."""
+    index = st.lists(st.text(min_size=1), min_size=2, max_size=10, unique=True)
+    cols = columns(
+        names_or_number=10,
+        elements=st.floats(allow_nan=False, allow_infinity=False),
+        unique=True,
```

Comment:
Not exactly sure if I understand your point.... These are all keyword args, so there should be no whitespace?! Or am I looking at the wrong spot?

Reply:
I think that @Scienfitz wants e.g. `names_or_number = 10` instead of `names_or_number=10`. I am however not aware of any regulations/rules which variant is preferable.

Reply:
But if that's what he means, then my answer is valid: this is a keyword argument --> there should be no whitespaces :D check literally ALL our remaining code or try to add them! Ruff will happily send them to trash after you've done so